src/ordered.jl

```diff
@@ -29,11 +29,11 @@ end
 
 Repeatedly tile a smaller matrix `mat` to fill out an image of height `h` and width `w`.
 """
-function tile_matrix(h, w, mat)
+function tile_matrix(mat, h, w)
     h_mat, w_mat = size(mat)
     repeat_rows = ceil(Int, h / h_mat)
     repeat_cols = ceil(Int, w / w_mat)
-    return repeat(mat, repeat_rows, repeat_cols)[1:h, 1:w] # trim to image dims
+    return view(repeat(mat, repeat_rows, repeat_cols), 1:h, 1:w) # view matching image dims
```

Comment:
I see two potential performance points here (not sure if it works)

- subarray(`view`) on a contiguous array is not contiguous, so multiplication with `tile_matrix` is not optimized.
- `repeat` allocates memory...

The better way is, perhaps, lazily repeating it and then using copy indexing to make a contiguous matrix as output.